extensions/ql-vscode/src/query-history/query-history-manager.ts

```diff
@@ -1177,57 +1178,40 @@ export class QueryHistoryManager extends DisposableObject {
   }
 
   private async findOtherQueryToCompare(
-    singleItem: QueryHistoryInfo,
-    multiSelect: QueryHistoryInfo[],
+    fromItem: CompletedLocalQueryInfo,
+    allItemsSelected: CompletedLocalQueryInfo[],
   ): Promise<CompletedLocalQueryInfo | undefined> {
-    // Variant analyses cannot be compared
-    if (
-      singleItem.t !== "local" ||
-      multiSelect.some((s) => s.t !== "local") ||
-      !singleItem.completedQuery
-    ) {
-      return undefined;
-    }
-    const dbName = singleItem.initialInfo.databaseInfo.name;
+    const dbName = fromItem.initialInfo.databaseInfo.name;
 
     // if exactly 2 queries are selected, use those
-    if (multiSelect?.length === 2) {
-      // return the query that is not the first selected one
-      const otherQuery = (
-        singleItem === multiSelect[0] ? multiSelect[1] : multiSelect[0]
-      ) as LocalQueryInfo;
-      if (!otherQuery.completedQuery) {
-        throw new Error("Please select a completed query.");
-      }
-      if (!otherQuery.completedQuery.successful) {
-        throw new Error("Please select a successful query.");
-      }
-      if (otherQuery.initialInfo.databaseInfo.name !== dbName) {
+    if (allItemsSelected.length === 2) {
+      const otherItem =
+        fromItem === allItemsSelected[0]
+          ? allItemsSelected[1]
+          : allItemsSelected[0];
```

Comment:
The previous code had a comment explaining a bit why we're doing this. That meaning is lost a little bit here. Is there something we can do to make it clear again? 

🤔 We could wrap this condition in a method like you've done in other places. Something like `firstUnselectedItem()`?